internal/bundle/plain/v0/bundle.go

```diff
@@ -0,0 +1,78 @@
+package v0
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apimacherrors "k8s.io/apimachinery/pkg/util/errors"
+	apimachyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const ManifestsPath = "manifests"
+
+var GVK = schema.GroupVersionKind{Group: "bundles.rukpak.io", Version: "v0", Kind: "Plain"}
```

Comment:
I like this structure of optional metadata file which embeds a typemeta. I can add this to the plain bundle spec as an optional source of metadata. Would we also want to manifests location field to the metadata, or are we still mandating that the manifests are under `/manifests`. I think we had added that because we didn't have a source of metadata defined -- if we do, maybe we could relax that assumption and allow the manifests to be in any specified directory?

Reply:
If there's a _really_ good use case for allowing the metadata to specify the location of the manifests, I think its something we could consider. But I feel like we haven't seen a use case for that with OLM's registry+v1 bundles (someone correct me if I'm wrong), so I think we should leave the `/manifests` directory location fixed for now.

At any point in the future we could add that to the metadata without a breaking change because we could say that `/manifests` is the default if it isn't defined. If we allow the manifest dir to be specified, there's no going back.